AutoGVP/select-clinVar-submissions.R

```diff
@@ -0,0 +1,176 @@
+#
+# NOTE: this script must be run BEFORE running run-autogvp.sh
+################################################################################
+
+suppressPackageStartupMessages({
+  library("tidyverse")
+  library("optparse")
+  library("vroom")
+})
+
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+
+## set up directories
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+analysis_dir <- file.path(root_dir, "AutoGVP")
+input_dir <- file.path(analysis_dir, "input")
+
+
+# parse parameters
+option_list <- list(
+  make_option(c("--variant_summary"),
+    type = "character",
+    help = "variant_summary file (format: variant_summary_2023-02.txt)"
+  ),
+  make_option(c("--submission_summary"),
+    type = "character",
+    help = "specific submission summary file (format: submission_summary.txt)"
+  )
+)
+
+opt <- parse_args(OptionParser(option_list = option_list))
+
+## get input files from parameters (reqd)
+input_submission_file <- opt$submission_summary
+input_variant_summary <- opt$variant_summary
+
+
+## load variant summary file, which reports latest clinVar consensus calls for each variant
+variant_summary_df <- vroom(input_variant_summary,
+  delim = "\t",
+  col_types = c(ReferenceAlleleVCF = "c", AlternateAlleleVCF = "c", PositionVCF = "i", VariationID = "n"),
+  show_col_types = FALSE
+) %>%
+  # retain only variants mapped to hg38
+  dplyr::filter(Assembly == "GRCh38" & ReferenceAlleleVCF != "na" & AlternateAlleleVCF != "na") %>%
+  # add vcf id column
+  dplyr::mutate(
+    vcf_id = str_remove_all(paste(Chromosome, "-", PositionVCF, "-", ReferenceAlleleVCF, "-", AlternateAlleleVCF), " "),
+    vcf_id = str_replace_all(vcf_id, "chr", ""),
+    VariationID = as.double(noquote(VariationID)),
+    LastEvaluated = case_when(
+      LastEvaluated == "-" ~ NA_character_,
+      TRUE ~ LastEvaluated
+    )
+  )
+
+# Load clinVar submission summary file, which reports all submissions for each clinVar variant
+submission_summary_df <- vroom(input_submission_file,
+  comment = "#", delim = "\t",
+  col_names = c(
+    "VariationID", "ClinicalSignificance", "DateLastEvaluated",
+    "Description", "SubmittedPhenotypeInfo", "ReportedPhenotypeInfo",
+    "ReviewStatus", "CollectionMethod", "OriginCounts", "Submitter",
+    "SCV", "SubmittedGeneSymbol", "ExplanationOfInterpretation"
+  ),
+  show_col_types = F
+)
+
+# remove submissions with missing columns (will have NA SubmittedGeneSymbol)
+submission_summary_df <- submission_summary_df[!is.na(submission_summary_df$SubmittedGeneSymbol), ]
+
+# Redefine `DateLastEvaluated`
+submission_summary_df <- submission_summary_df %>%
+  dplyr::mutate(DateLastEvaluated = case_when(
+    DateLastEvaluated == "-" ~ NA_character_,
+    TRUE ~ DateLastEvaluated
+  ))
+
+# merge submission_summary and variant_summary info
+submission_merged_df <- submission_summary_df %>%
+  dplyr::rename("LastEvaluated" = DateLastEvaluated) %>%
+  left_join(variant_summary_df, by = c("VariationID")) %>%
```

Comment:
```suggestion
  left_join(variant_summary_df, by = "VariationID", suffix = c("_sub", "_var")) %>%
```
Don't need the `c()` if only one variable. Also, I prefer not to see `.x` and `.y` in column names, but as you are using these below, you can add a suffix to these files to denote their origin, making it easier for the user/code reviewer to follow. This will require code adjustment below to update the column names, but this way it is cleaner and clearer.